src/globus_cli/commands/cli_profile_list.py

```diff
@@ -97,7 +97,7 @@ def cli_profile_list(*, all: bool) -> None:
         ]
         if all:
             fields.append(Field("GLOBUS_SDK_ENVIRONMENT", "env"))
-        display(user_profiles, text_mode=TextMode.text_table, fields=fields)
+        display(user_profiles, text_mode=display.TABLE, fields=fields)
```

Comment:
The default is to display a table. Does every call site specify the `text_mode` to use? If so, I suggest removing the default parameter value.

Reply:
I was thinking about this when I made the change. It may require a few touches, but I can't imagine many commands are relying on the default. Let's get rid of that and make it a required keyword-only param.

Reply:
Actually, it seems there are a few, but this is raising a separate interface improvement -- I think we can eliminate `simple_text` with a little bit of extra elbow-grease. Given that it complicates this interface, I'm going to give that cleanup a shot this evening.

Reply:
To explain how it relates, here's how it came up when I was looking at `globus whoami`:
```diff
@@ -81,11 +83,7 @@ def whoami_command(login_manager: LoginManager, *, linked_identities: bool) -> N
                     Field("ID", "sub"),
                     Field("Email", "email"),
                 ],
-                simple_text=(
-                    None
-                    if is_verbose()
-                    else "\n".join([x["username"] for x in res["identity_set"]])
-                ),
+                text_mode=display.TABLE if is_verbose() else _render_linked_identities,
             )
         except KeyError:
             click.echo(
@@ -99,12 +97,20 @@ def whoami_command(login_manager: LoginManager, *, linked_identities: bool) -> N
     else:
         display(
             res,
-            text_mode=display.RECORD,
+            text_mode=display.RECORD if is_verbose() else _render_preferred_username,
             fields=[
                 Field("Username", "preferred_username"),
                 Field("Name", "name"),
                 Field("ID", "sub"),
                 Field("Email", "email"),
             ],
-            simple_text=(None if is_verbose() else res["preferred_username"]),
         )
+
+
+def _render_linked_identities(data: t.Any) -> None:
+    for x in data["identity_set"]:
+        click.echo(x["username"])
+
+
+def _render_preferred_username(data: t.Any) -> None:
+    click.echo(data["preferred_username"])
```

Reply:
Per my last comment, the attempted change, and the revert, are we okay punting this problem out of the scope of this PR?

I don't want to make it a required param because then usages with `simple_text` will have to specify it _and_ override it. I haven't noodled out the right interface fix for all of this yet.

For now, I want to stick with reducing the number of symbols being passed around (`display, TextMode` -> `display`) and making `display` into a callable object which opens up some interesting options for giving it properties or otherwise being a "smarter" implementation.